test/unit/server/handlers/consents/{ID}/generateChallenge.test.ts

```diff
@@ -0,0 +1,115 @@
+import ConsentsIdGenerateChallengeHandler from '~/server/handlers/consents/{ID}/generateChallenge'
+
+const mockForwardConsentsIdGenerateChallengeRequest = jest.spyOn(ConsentIdGenerateChallengeRequests, 'forwardConsentsIdGenerateChallengeRequest')
+const mockLoggerPush = jest.spyOn(Logger, 'push')
+const mockLoggerError = jest.spyOn(Logger, 'error')
+
+const mockPostConsentsIdGenerateChallengeRequest = {
+  headers: {
+    'fspiop-source': 'pispA',
+    'fspiop-destination': 'auth.dfspa'
+  },
+  params: {
+    ID: 'b82348b9-81f6-42ea-b5c4-80667d5740fe'
+  },
+  payload: {
+    type: 'FIDO'
+  }
+}
+
+const mockForwardConsentsIdGenerateChallengeRequestExpectedWeb = [
+  'b82348b9-81f6-42ea-b5c4-80667d5740fe',
+  '/consents/{{ID}}/generateChallenge',
+  'TP_CB_URL_CONSENT_GENERATE_CHALLENGE_POST',
+  mockPostConsentsIdGenerateChallengeRequest.headers,
+  'POST',
+  mockPostConsentsIdGenerateChallengeRequest.payload,
+  undefined
+]
+
+describe('consentsIdGenerateChallenge handler', () => {
+  describe('POST /consents/{ID}/generateChallenge', () => {
+    beforeEach((): void => {
+      jest.clearAllMocks()
+      mockLoggerPush.mockReturnValue(null)
+      mockLoggerError.mockReturnValue(null)
+    })
+
+    it('handles a successful request', async () => {
+      // Arrange
+      mockForwardConsentsIdGenerateChallengeRequest.mockResolvedValueOnce()
+      const request = mockPostConsentsIdGenerateChallengeRequest
+      const expected = mockForwardConsentsIdGenerateChallengeRequestExpectedWeb
+
+      // Act
+      const response = await ConsentsIdGenerateChallengeHandler.post(null, request as unknown as Request, mockResponseToolkit)
+
+      // Assert
+      expect(response.statusCode).toBe(202)
+      expect(mockForwardConsentsIdGenerateChallengeRequest).toHaveBeenCalledWith(...expected)
+    })
+
+    it('handles errors asynchronously', async () => {
+      // Arrange
+      mockForwardConsentsIdGenerateChallengeRequest.mockRejectedValueOnce(new Error('Test Error'))
+      const request = mockPostConsentsIdGenerateChallengeRequest
+      const expected = mockForwardConsentsIdGenerateChallengeRequestExpectedWeb
+
+      // Act
+      const response = await ConsentsIdGenerateChallengeHandler.post(null, request as unknown as Request, mockResponseToolkit)
+
+      // Assert
+      expect(response.statusCode).toBe(202)
+      // wait once more for the event loop - since we can't await `runAllImmediates`
+      // this helps make sure the tests don't become flaky
+      await new Promise(resolve => setImmediate(resolve))
+      // The main test here is that there is no unhandledPromiseRejection!
+      expect(mockForwardConsentsIdGenerateChallengeRequest).toHaveBeenCalledWith(...expected)
+    })
+
+    it('handles validation errors synchronously', async () => {
+      // Arrange
+      const request = {
+        ...mockPostConsentsIdGenerateChallengeRequest,
+        // Will setting the span to null do stuff?
```

Comment:
nit: can we please remove this copy paste?

Reply:
let me leave this for another PR.